### DISCUSSION TITLE
GLMsingle / GLMdenoise in pRF designs?

### DISCUSSION THREAD

**Discussion by benharvey** [cvnlab/GLMsingle#78] [Q&A] [answered]
Hi Kendrick,
is GLMdenoise (or its implementation in GLMsingle) compatible with pRF modelling experimental designs? I can find mentions of block and event-related designs, but nothing about pRF designs. It strikes me that a pRF model is a bit like a block design, a bit like event-related, but not really either. And there may be complications because pRF analyses test the predictions of many candidate response function predictions. Nevertheless, I feel like could easily identify non-responsive areas, estimate the noise regressors, and regress them out globally. Has anyone ever tried this, or is it doomed to fail?

Cheers,
Ben

**Comment by kendrickkay** [comment, marked as answer]
Hi Ben - This is a good and (hard) question. I thought/considered/tried things along these lines in the past (maybe 7 years ago??). 

Here are my current thoughts from that exploration:
- I agree with your assessment that it's sort of like block designs (if you use a "slowly moving" type stimulus design), but statistically a little different (assuming you use a standard nonlinear optimization fitting approach).
- Empirically, I found that some implementations of the general idea really didn't change the "experimental response" estimates much at all.  The changes were almost imperceptible.  This could possibly be due to the fact that the datasets I was looking at had great SNR to start with (and/or very little correlated noise artifacts), or the fact that the types of stimulus-driven responses in pRF paradigms do not correlate much with fMRI noise sources, or both.
- Technically, implementing the GLMdenoise-like noise procedure is a bit complicated and it felt more and more like it just wasn't worth the complications.

That being said, if you had a specific type of data / paradigm that is clearly corrupted by noise artifacts, it might be perhaps worthwhile to revisit these thoughts.

Perhaps you thought about other related but different denoising approaches (e.g. ICA based method)?  There are major potential risks, certainly, of inducing biases (removing signals).  In fact, there's a little paper on this general topic :)   --> https://doi.org/10.1371/journal.pone.0270895

Kendrick

**Reply by kendrickkay** [reply]
PS: If you use a "static/randomized" pRF mapping design, then application of GLMsingle is easy and straightforward. Of course, there are pros and cons of such designs compared to continuous more block-like designs, but I generally like the "event-related" designs for the types of scientific questions I am interested in...

**Comment by benharvey** [comment]
Thanks Kendrick,
that's all very useful. I thought it would have crossed your mind. I am interested in denoising for non-spatial pRF models, things like tuning for event timing and numerosity, where the signals are weaker. There aren't clear artefacts here, just a need for a lot of runs to see clear responses.
But I appreciate that it is not as simple as in block and event-related designs, and you make some good points about introducing biases. I think I'll stop thinking about it. NORDIC denoising has been a great help already.

Ben

**Comment by kendrickkay** [comment]
I see.  I suppose you should decide for yourself what types of noise are probably most prevalent (as that will determine the compensation strategy, if any, you wish to pursue).  Thermal noise (e.g. if you are imaging at very high resolution)? Head motion related noise? Physiological noise? Cognitive/behavior noise?    (PS: Note that NORDIC may be introducing bias (as discussed in the paper)...  essentially it (obviously) introduces complex dependencies across the voxels considered in each window...  so I would suggest to proceed with caution.)
